templates/srpm.erb

```diff
@@ -0,0 +1,233 @@
+<%# avoids a lot of potential errors, including the case that -%>
+<%# the script is non-empty, but just whitespace and/or comments -%>
+    :
+<%     if script?(:after_upgrade) -%>
+<%=      script(:after_upgrade) %>
+<%     end -%>
+}
+_install() {
+<%# Making sure that at least one command is in the function -%>
+<%# avoids a lot of potential errors, including the case that -%>
+<%# the script is non-empty, but just whitespace and/or comments -%>
+    :
+<%     if script?(:after_install) -%>
+<%=      script(:after_install) %>
+<%     end -%>
+}
+if [ "${1}" -eq 1 ]
+then
+    # "after install" goes here
+    _install
+elif [ "${1}" -gt 1 ]
+then
+    # "after upgrade" goes here
+    upgrade
+fi
+<%   end -%>
+<%   if script?(:before_remove) -%>
+%preun <% if attributes[:rpm_macro_expansion?] -%><%= " -e " %>  <% end %>
+if [ "${1}" -eq 0 ]
+then
+<%# Making sure that at least one command is in the function -%>
+<%# avoids a lot of potential errors, including the case that -%>
+<%# the script is non-empty, but just whitespace and/or comments -%>
+    :
+<%=    script(:before_remove) %>
+fi
+<%   end -%>
+<%   if script?(:after_remove) -%>
+%postun <% if attributes[:rpm_macro_expansion?] -%><%= " -e " %>  <% end %>
+if [ "${1}" -eq 0 ]
+then
+<%# Making sure that at least one command is in the function -%>
+<%# avoids a lot of potential errors, including the case that -%>
+<%# the script is non-empty, but just whitespace and/or comments -%>
+    :
+<%=    script(:after_remove) %>
+fi
+<%   end -%>
+<% else
+     other_scriptmap = {
+       :before_install => "pre",
+       :after_install => "post",
+       :before_remove => "preun",
+       :after_remove => "postun"
+     }
+     scriptmap.merge!(other_scriptmap)
+   end
+-%>
+<% scriptmap.each do |name, rpmname| -%>
+<%   if script?(name) -%>
+%<%=   rpmname -%> <%= ' -e' if attributes[:rpm_macro_expansion?] %>
+<%=    script(name) %>
+<%   end -%>
+<% end -%>
+
+<%# This section adds any triggers, as ordered in the command line -%>
+<%
+  triggermap = {
+    :before_install => "prein",
+    :after_install => "in",
+    :before_uninstall => "un",
+    :after_target_uninstall => "postun"
+  } 
+  triggermap.each do |name, rpmtype|
+    (attributes["rpm_trigger_#{name}".to_sym] or []).each do |trigger_name, trigger_script, trigger_scriptprog| -%>
+%trigger<%= rpmtype -%> <%= trigger_scriptprog -%> -- <%= trigger_name %>
+<%= trigger_script %>
+<%   end -%>
+<% end -%>
+
+%files
+%defattr(<%= attributes[:rpm_defattrfile] %>,<%= attributes[:rpm_user] || "root" %>,<%= attributes[:rpm_group] || "root" %>,<%= attributes[:rpm_defattrdir] %>)
+<%# Output config files and then regular files. -%>
+<% config_files.each do |path| -%>
+%config(noreplace) <%= rpm_file_entry(path) %>
+<% end -%>
+<%# list directories %>
+<% directories.each do |path| -%>
+%dir <%= rpm_file_entry(path) %>
+<% end -%>
+<%# list only files, not directories? -%>
+/
```

Comment:
Since we're building an rpm from source with rpmbuild, we might be able to get away with listing `/` as the %files entry and including all files that end up in the %buildroot.

This is mostly a hack at this time to avoid having to do additional file list generation in this prototype.